store/sqlstore/post_store.go

```diff
@@ -1228,7 +1228,9 @@ func (s *SqlPostStore) getPostsSinceCollapsedThreads(options model.GetPostsSince
 		Where(sq.Eq{"Posts.ChannelId": options.ChannelId}).
 		Where(sq.Gt{"Posts.UpdateAt": options.Time}).
 		Where(sq.Eq{"Posts.RootId": ""}).
-		OrderBy("Posts.CreateAt DESC").ToSql()
+		OrderBy("Posts.CreateAt DESC").
+		Limit(1000).
```

Comment:
1/5, can we get the limit from the options? For example: 

```
uint64(options.PerPage)
```

Reply:
Yep, I agree - the call stack for this function from the API layer is:

`api4.getPostsForChannel` -> `app.GetPostsSince` -> `store.GetPostsSince` -> `poststore.getPostsSinceCollapsedThreads` (this function)

In the API layer (getPostsForChannel), we have the `page` and `perPage` params. These params are automatically set to the correct maxes in https://github.com/mattermost/mattermost-server/blob/master/web/params.go by the handler logic.

We should just pass those params down all the way down to this function, similar to how its done for `Store.GetPosts`

Reply:
It seems intentional to me to fetch the maximum posts when using the `since` route, as both server and client don't pass limit to it, all the while passing it to the other routes like `after` within the same code.
Client - https://github.com/mattermost/mattermost-webapp/blob/19bbedfc96f6ac8c5018bfb39b804f40a621cd45/packages/client/src/client4.ts#L1986-L2005

[Server PR](https://github.com/mattermost/mattermost-server/pull/5858/files#diff-6d5f868d32140b9df8b7767cf3c967114ee2f8d48f99bc60e73bec01e25b1a5d) seemed to have done this intentionally.

Also the default limit(60) would be much smaller than the current limit(1000), I'm not sure if this may break things.

Reply:
@srkgupta @ashishbhate LMK, if you have strong opinion on it, I can certainly pass the two params, otherwise we can keep it this way.

Reply:
>It seems intentional to me to fetch the maximum posts when using the since route, as both server and client don't pass limit to it...

I think that's exactly what we need to fix here. 

>[Server PR](https://github.com/mattermost/mattermost-server/pull/5858/files#diff-6d5f868d32140b9df8b7767cf3c967114ee2f8d48f99bc60e73bec01e25b1a5d) seemed to have done this intentionally.

That PR is very old, it isn't a good example for how things should be now. 

>Also the default limit(60) would be much smaller than the current limit(1000), I'm not sure if this may break things.

Isn't 1000 an arbitrary number? It could just as well break things, and/or cause resource exhaustion.

It seems to me the correct fix for this ticket is to fix the clients to use pagination correctly, and to use the proper default limits here, instead of an arbitrary limit. 


If this is urgent then we can merge as is, and create a follow-up jira ticket for a better overall fix. This could be a good candidate for the upcoming code quality sprint. 

cc @Willyfrog

Reply:
> It seems to me the correct fix for this ticket is to fix the clients to use pagination correctly, and to use the proper default limits here, instead of an arbitrary limit.

Clients can specify their limit to prevent breaking, but there should be a max and a default. Do we have those defined for other instances (e.g.: where the `since` parameter is not used)?

> If this is urgent then we can merge as is, and create a follow-up jira ticket for a better overall fix. This could be a good candidate for the upcoming code quality sprint.

I'm ok with this if it helps prevent the security issue. And I agree it is a good candidate for that. Keep in mind if we are fixing the clients as well, it might need help from other team members (which is ok, just surfacing it)

Reply:
> Clients can specify their limit to prevent breaking, but there should be a max and a default. Do we have those defined for other instances (e.g.: where the since parameter is not used)?

Yes, we do. It is what @ashishbhate points to in the first comment: https://github.com/mattermost/mattermost-server/blob/aef5165cdc7d302fff589204b23a48e7a6753d54/web/params.go#L19-L20

> It seems to me the correct fix for this ticket is to fix the clients to use pagination correctly, and to use the proper default limits here, instead of an arbitrary limit.

This is the right fix. It was a mistake for the server to return everything and for the client to assume that everything will be returned. The client should paginate.